### DISCUSSION TITLE
Training a Multisession Model And Applying To Multiple Different Mice

### DISCUSSION THREAD

**Discussion by gbradham** [AdaptiveMotorControlLab/CEBRA#217] [Q&A] [answered]
Hello,

First I would like to say thank you very much for this amazing tool.

After reading and looking at the demo notebook, I saw that CEBRA would allow for the training of multiple mice with varying neuron # and then show embeddings. I was successful able to train the model and plot the embeddings, but if I wanted to apply this model to a 'test' dataset of mice, it would not work because of the varying neuron #.

I am fairly new to all of this, and I believe that I may be doing this wrong or do not understand what I need to be doing. Below is the error I receive when trying to transform the model to a new mouse.
`multi_cebra_model.transform(datas_test[0], session_id=i)`
![Screenshot from 2025-01-22 16-55-24](https://github.com/user-attachments/assets/5c456afb-f256-4b28-89e1-05731ac2a22c)

I train off of a neural data csv (Shape: (3000, X where X is the neuron count so each neuron has its own column)
And off of a grooming data csv (Shape: (3000, 1 where the one column is 0/1 where 0 is not grooming and 1 is grooming.
I have a train set of 5 mice and a test set of 3 mice.

Is CEBRA restricted to only the same neuron # per the session it was trained on? What approach should I take to be able to use the CEBRA model to predict a 0(not grooming) or 1(grooming) on my test mice?

Thank you again for CEBRA and I will be swift to answer if more details need to be provided!

**Comment by icarosadero** [comment]
If you are planning to use the same embedding for another experiment (another mice), you must make sure that the number of channels (neurons) are the same. In the error message, it seems that you used 21 channels to train, but asked the model to work on only 12. A possible workaround could be to pad your data with zeroes before, so that in all cases you have the same number of channels. At least that is what I'm doing for now, and I am not entirely sure if it really works.

**Reply by gbradham** [reply]
Thank you for the reply. I will definitely try this.

**Comment by MMathisLab** [comment, marked as answer]
the best way to do this is to use `Adapt = true`; this takes a pretrained embedding and adapts the first input layer so it matches the neuron # appropriately.

**Reply by gbradham** [reply]
Thank you so much!!
